script.js

```diff
@@ -0,0 +1,338 @@
+  }
+
+  static generateKeydownEvent(code) {
+    const event = new KeyboardEvent('keydown', {
+      bubbles: true,
+      cancelable: true,
+      keyCode: code,
+    });
+    document.dispatchEvent(event);
+  }
+
+  static highlightKey(key) {
+    key.classList.add('active');
+    setTimeout(() => {
+      key.classList.remove('active');
+    }, 300);
+  }
+
+  createKeyboard(btns) {
+    for (let i = 0; i < btns.length; i += 1) {
+      this.keyboard.append(KeyBoard.createKeyElement(btns[i]));
+    }
+
+    document.querySelectorAll('.key').forEach((item) => {
+      item.addEventListener('mousedown', () => {
+        KeyBoard.highlightKey(item);
+        KeyBoard.generateKeydownEvent(item.dataset.code);
+      });
+    });
+  }
+
+  changeCase() {
+    this.keyboard.innerHTML = '';
+    switch (this.activeKeyboard) {
+      case enLowerCase:
+        this.createKeyboard(enUpperCase);
+        this.activeKeyboard = enUpperCase;
+        break;
+
+      case enUpperCase:
+        this.createKeyboard(enLowerCase);
+        this.activeKeyboard = enLowerCase;
+        break;
+
+      case ruLowerCase:
+        this.createKeyboard(ruUpperCase);
+        this.activeKeyboard = ruUpperCase;
+        break;
+
+      case ruUpperCase:
+        this.createKeyboard(ruLowerCase);
+        this.activeKeyboard = ruLowerCase;
+        break;
+
+      default:
+        this.createKeyboard(enLowerCase);
+        this.activeKeyboard = enLowerCase;
+    }
+  }
+
+  changeLang() {
+    this.keyboard.innerHTML = '';
+    switch (this.activeKeyboard) {
+      case enLowerCase:
+        this.createKeyboard(ruLowerCase);
+        this.activeKeyboard = ruLowerCase;
+        localStorage.setItem('myKeyboardLang', '"ru"');
+        break;
+
+      case enUpperCase:
+        this.createKeyboard(ruUpperCase);
+        this.activeKeyboard = ruUpperCase;
+        localStorage.setItem('myKeyboardLang', '"ru"');
+        break;
+
+      case ruLowerCase:
+        this.createKeyboard(enLowerCase);
+        this.activeKeyboard = enLowerCase;
+        localStorage.setItem('myKeyboardLang', '"en"');
+        break;
+
+      case ruUpperCase:
+        this.createKeyboard(enUpperCase);
+        this.activeKeyboard = enUpperCase;
+        localStorage.setItem('myKeyboardLang', '"en"');
+        break;
+
+      default:
+        this.createKeyboard(enLowerCase);
+        this.activeKeyboard = enLowerCase;
+        localStorage.setItem('myKeyboardLang', 'en');
+    }
+  }
+
+  insertKeyboard(parent) {
+    this.keyboard.classList.add('keyboard');
+
+    const textarea = document.createElement('textarea');
+    textarea.setAttribute('autofocus', '');
+
+    parent.append(textarea);
+    parent.append(this.keyboard);
+
+    this.activeKeyboard = this.lang === 'en' ? enLowerCase : ruLowerCase;
+    this.createKeyboard(this.activeKeyboard);
+
+    document.addEventListener('keydown', (e) => {
+      e.preventDefault();
+      document.querySelectorAll('.key').forEach((item) => {
```

Comment:
This is a redundant querySelector and loop. On every keydown event you are looping throw all elements.It's overloaded actions. You have a data-code on each element by which you can get needed one and use it.
```javascript
const item = document.querySelector(`[data-code="${e.keyCode}"]`);
```
And the second condition in 'if' is useless as well `e.keyCode === parseInt(item.dataset.code, 10))`